damnit/ctxsupport/ctxrunner.py

```diff
@@ -594,6 +598,89 @@ def save_hdf5(self, hdf5_path, reduced_only=False):
         if os.stat(hdf5_path).st_uid == os.getuid():
             os.chmod(hdf5_path, 0o666)
 
+
+@contextmanager
+def filesystem(host='localhost'):
+    """Mount a remote filesystem with sshfs"""
+    if host == 'localhost':
+        yield
+        return
+
+    with TemporaryDirectory() as td:
+        try:
+            run(f'sshfs {host}:{ed.read_machinery.DATA_ROOT_DIR} {td}', check=True, shell=True)
```

Comment:
Will this ever prompt the user for credentials?

Reply:
Ok, sounds good :+1: I dunno if it's possible, but is there a way to get `sshfs` to exit with an error if it can't authenticate without a prompt? That would make debugging a bit easier if we ever get into that situation.

Reply:
yea we can pass ssh options, so we can deactivate password authentication:

```shell
sshfs ... -o ssh_command='ssh -o PasswordAuthentication=no'
```